static/run-snapd-from-snap: adapt to new mount place for snapd snap

alfonsosanchezbeato opened the pull request:
Now we will mount the snapd snap in /snap/snapd/<revision> on first boot to avoid duplicate mounts. Adapt to this initramfs changes.

alfonsosanchezbeato commented:
This would need to be backported to all core20+ bases. The change is compatible with older snap-bootstrap and with the new one with the changes in https://github.com/canonical/snapd/pull/14548.

valentindavid commented on static/usr/lib/core/run-snapd-from-snap:
```diff
@@ -5,31 +5,40 @@
+        # dynamic linker
+        # /snap/snapd/current/usr/lib/x86_64-linux-gnu/ld-linux-x86-64.so.2
+        # is available to run snapd.
+        [ -d /snap/snapd ] || mkdir -p /snap/snapd
```
Think about it, the condition was useless here, right? Maybe you can remove it in the same time.

alfonsosanchezbeato replied:
Good point, removed now.

Meulengracht reviewed (approved):
lgtm, thank you

alfonsosanchezbeato commented:
I've rebased this now and added some minor changes to be more flexible as we might have different listening socket in the future - asking for a re-review.

valentindavid reviewed (approved):
Looks fine. Just one question about the multiple sockets (you did not add it in this PR, but I am still wondering).
